src/services/refactors/convertReactPureComponent.ts

```diff
@@ -0,0 +1,294 @@
+     * 2. [x] If the selection is a function declaration of type `React.SFC`
+     *        [ ] Provide an action to convert it to `React.PureComponent`
+     *        [ ] * Make a function with 1 to 2 arguments (treat as props & context)
+     *              and explicit returnType T and T subtypeable to React.Element also convertible
+     * 3. [x] If the selection is a subclass of `React.Component` or `React.PureComponent`
+     *        [x] and if it satisifies `isConvertableClass()`
+     *        [ ] Provide an action to convert it to `React.SFC` or `React.PureComponent` (if it is `React.Component`)
+     * 4. [x] Resolve React by `getReferenceToReact()`
+     * 5. [ ] * Also support `propTypes`, `contextTypes`, `defaultProps`, `displayName` (Only Class -> Function)
+     * 6. [ ] * Also support convert of reference to `this.context` (in both ways)
+     */
+
+    registerRefactor(refactorName, {
+        getAvailableActions(context: RefactorContext): ApplicableRefactorInfo[] | undefined {
+            // if (!checkJSXFactoryIsReact(context.host)) return;
+            if (!getReferenceToReact(context.file)) return;
+            const node = getSelectedNode(context);
+            if (!node) { return; }
+
+            if (isConvertibleReactClassComponent(node, context.file)) {
+                // const description = Diagnostics.???
+                const description = actionNameComponentToSFC;
+                return [{
+                    name: refactorName,
+                    description,
+                    actions: [{ description, name: actionNameComponentToSFC }]
+                }];
+            }
+            else if (isReactSFCDeclaration(node, context.file)) {
+                // const description = Diagnostics.???
+                const description = actionNameSFCToPureComponent;
+                return [{
+                    name: refactorName,
+                    description,
+                    actions: [{ description, name: actionNameSFCToPureComponent }]
+                }];
+            }
+        },
+        getEditsForAction(context: RefactorContext, actionName: string): RefactorEditInfo | undefined {
+            Debug.assert(actionName === actionNameComponentToSFC || actionName === actionNameSFCToPureComponent);
+            const node = getSelectedNode(context) as Node;
+            if (actionName === actionNameSFCToPureComponent) {
+                return transformSFCtoComponent(isReactSFCDeclaration(node, context.file)!, context);
+            }
+            else if (actionName === actionNameComponentToSFC) {
+                return transformComponentToSFC(isConvertibleReactClassComponent(node, context.file)!, context);
+            }
+            return;
+        }
+    });
+
+    // function checkJSXFactoryIsReact(host: LanguageServiceHost) {
+    //     const config = host.getCompilationSettings();
+    //     if (config.jsxFactory === undefined) { if (config.jsx) return true; }

```

Comment:
as i noted erlier, you do not want to limit this to React or the module name "react" there is peact, vue, mithril, etc..

Reply:
But this refactor is meaningless for the Vue and Mithril, why don't we limit it to React, ReactNative and Preact?